source/fundamentals/connection.txt

```diff
@@ -245,6 +245,16 @@ that Context deadline for the operation. If the context does not have a
 deadline, the driver derives a new Context from the given Context using
 the ``Timeout`` option set on the ``Client``.
 
+.. note:: Retryability
+
+   This version of the {+driver-short+} enables retryable reads and
+   writes by default. If you set a ``Timeout`` option on your ``Client``
+   and your operation requires a retry, the driver retries the operation
+   as many times as possible before the timeout expires. Once the timeout
```

Comment:
Suggestion:
Mention that you can use the maxConnecting parameter to rate limit the new connection attempts if they observe the server getting flooded.

As a reader, when I saw "as many times as possible", I was concerned that this might flood a connection until I read "[Why don't drivers use backoff/jitter between retry attempts"](https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#why-dont-drivers-use-backoff-jitter-between-retry-attempts) in the spec.

If that detail doesn't seem important, then feel free not to mention anything. I'm also not certain whether maxConnecting is a Timeout option.

Reply:
There is a Client option in the Go Driver to set `MaxConnecting`, but I'm not sure that mentioning this detail relates directly to the note. Setting `MaxConnecting` seems to be a safeguard when multiple concurrent operations select the same server. The spec doesn't seem to recommend setting `MaxConnecting` in general.

Reply:
I just thought it might help address an engineer's concern about connection floods/storms because retry as many times as fast as possible isn't an expected connection behavior. As mentioned in the linked doc, there's usually backoff in which it slows down the retries after each failure. Fine with omitting info about MaxConnecting.